pkg/controller/realmadmin/settings.go

```diff
@@ -121,10 +121,18 @@ func (c *Controller) HandleSettings() http.Handler {
 		if form.Codes {
 			realm.AllowedTestTypes = form.AllowedTestTypes
 			realm.RequireDate = form.RequireDate
-			realm.CodeLength = form.CodeLength
-			realm.CodeDuration.Duration = time.Duration(form.CodeDurationMinutes) * time.Minute
-			realm.LongCodeLength = form.LongCodeLength
-			realm.LongCodeDuration.Duration = time.Duration(form.LongCodeDurationHours) * time.Hour
+			if v := realm.CodeLength; v > 0 {
```

Comment:
better to just check if ENX is enabled and skip the individual checks?

Reply:
Donie